huaweicloud/services/live/resource_huaweicloud_live_snapshot.go

```diff
@@ -0,0 +1,335 @@
+		var (
+			updateLiveSnapshotHttpUrl = "v1/{project_id}/stream/snapshot"
+			updateLiveSnapshotProduct = "live"
+		)
+		updateLiveSnapshotClient, err := cfg.NewServiceClient(updateLiveSnapshotProduct, region)
+		if err != nil {
+			return diag.Errorf("error creating Live Client: %s", err)
+		}
+
+		updateLiveSnapshotPath := updateLiveSnapshotClient.Endpoint + updateLiveSnapshotHttpUrl
+		updateLiveSnapshotPath = strings.ReplaceAll(updateLiveSnapshotPath, "{project_id}",
+			updateLiveSnapshotClient.ProjectID)
+
+		updateLiveSnapshotOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+			OkCodes: []int{
+				200,
+			},
+		}
+		updateLiveSnapshotOpt.JSONBody = utils.RemoveNil(buildUpdateLiveSnapshotBodyParams(d, region))
+		_, err = updateLiveSnapshotClient.Request("PUT", updateLiveSnapshotPath, &updateLiveSnapshotOpt)
+		if err != nil {
+			return diag.Errorf("error updating Live snapshot: %s", err)
+		}
+	}
+	return resourceLiveSnapshotRead(ctx, d, meta)
+}
+
+func buildUpdateLiveSnapshotBodyParams(d *schema.ResourceData, region string) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"domain":            utils.ValueIngoreEmpty(d.Get("domain_name")),
+		"app_name":          utils.ValueIngoreEmpty(d.Get("app_name")),
+		"time_interval":     utils.ValueIngoreEmpty(d.Get("frequency")),
+		"object_write_mode": utils.ValueIngoreEmpty(d.Get("storage_mode")),
+		"obs_location":      buildLiveSnapshotObsLocationChildBody(d, region),
+		"call_back_enable":  utils.ValueIngoreEmpty(d.Get("call_back_enabled")),
+		"call_back_url":     utils.ValueIngoreEmpty(d.Get("call_back_url")),
+		"auth_key":          utils.ValueIngoreEmpty(d.Get("call_back_auth_key")),
+	}
+	return bodyParams
+}
+
+func resourceLiveSnapshotDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// deleteLiveSnapshot: Delete Live snapshot
+	var (
+		deleteLiveSnapshotHttpUrl = "v1/{project_id}/stream/snapshot"
+		deleteLiveSnapshotProduct = "live"
+	)
+	deleteLiveSnapshotClient, err := cfg.NewServiceClient(deleteLiveSnapshotProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating Live Client: %s", err)
+	}
+
+	deleteLiveSnapshotPath := deleteLiveSnapshotClient.Endpoint + deleteLiveSnapshotHttpUrl
+	deleteLiveSnapshotPath = strings.ReplaceAll(deleteLiveSnapshotPath, "{project_id}",
+		deleteLiveSnapshotClient.ProjectID)
+
+	deleteLiveSnapshotDeleteParams := buildDeleteLiveSnapshotDeleteParams(d)
+	deleteLiveSnapshotPath += deleteLiveSnapshotDeleteParams
+
+	deleteLiveSnapshotOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	_, err = deleteLiveSnapshotClient.Request("DELETE", deleteLiveSnapshotPath, &deleteLiveSnapshotOpt)
+	if err != nil {
+		return diag.Errorf("error deleting Live snapshot: %s", err)
+	}
+
+	return nil
+}
+
+func buildDeleteLiveSnapshotDeleteParams(d *schema.ResourceData) string {
```

Comment:
`dimain_name` and `app_name` is forceNew and required. Ithink we can remove the `d.GetOK()` check. 
Just return `return fmt.Sprintf("?domain=%v&app_name=%v", d.Get("domain_name"), d.Get("app_name"))`

Reply:
fixed